src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Optional<Role> getRoleByName(Role.RoleName roleName) {
+        try (Session session = factory.openSession()) {
+            Query<Role> getRoleByName = session.createQuery(
+                    "FROM Role WHERE roleName = :roleName", Role.class);
+            getRoleByName.setParameter("roleName", roleName);
+            return getRoleByName.uniqueResultOptional();
+        } catch (Exception e) {
+            throw new DataProcessingException("No found a role for the user " + roleName + e);
```

Comment:
```suggestion
            throw new DataProcessingException("Cant get Role by RoleName " + roleName, e);
```

We get this esception if smth went wrong while we getting Role from db